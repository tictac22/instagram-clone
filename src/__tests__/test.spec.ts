import { beforeEach, describe, expect, it } from "vitest"

import { mount, VueWrapper } from "@vue/test-utils"
import Test from "../components/test.vue"

const enum dataAttrs {
	counter = '[data-test="counter"]',
	increment = '[data-test="increment"]',
	decrement = '[data-test="decrement"]'
}
describe("test component", () => {
	let component: VueWrapper
	beforeEach(() => {
		component = mount(Test)
	})
	it("It mounts succesfullyy ", () => {
		const counter = component.find(dataAttrs.counter)
		expect(counter.text()).toContain("counter 0")
		expect(component.element).toMatchSnapshot()
	})
	it("it increments correctly", async () => {
		await component.find(dataAttrs.increment).trigger("click")
		const counter = component.find(dataAttrs.counter)
		expect(counter.text()).toContain("counter 1")
	})
	it("it decrements correctly", async () => {
		await component.find(dataAttrs.decrement).trigger("click")
		const counter = component.find(dataAttrs.counter)
		expect(counter.text()).toContain("-1")
	})
})
