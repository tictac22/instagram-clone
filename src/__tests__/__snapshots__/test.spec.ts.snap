// Vitest Snapshot v1

exports[`test component > It mounts succesfullyy  1`] = `
<div>
  <h1
    data-test="counter"
  >
    counter 0
  </h1>
  <button
    data-test="increment"
  >
    increment
  </button>
  <button
    data-test="decrement"
  >
    decrement
  </button>
</div>
`;
